dcc-portal-ui/app/scripts/common/js/services.js

```diff
@@ -347,4 +347,32 @@
     }
   });
 
+  module.service('EnsureInputService', function(){
```

Comment:
All of these look highly suspicious in terms of reuse. My suggestion would be to inline them if they are only used in a single file.